common/remote/sockfile.cpp

```diff
@@ -5637,6 +5869,179 @@ class CRemoteFileServer : implements IRemoteFileServer, public CInterface
+    {
+        // this is an attempt to authenticate when we haven't got authentication turned on
+        if (TF_TRACE_CLIENT_STATS)
+        {
+            StringBuffer s(client.queryPeerName());
+            PROGLOG("Connect from %s",s.str());
+        }
+
+        Owned<IPropertyTree> requestTree = createPTreeFromJSONString(msg.length(), msg.toByteArray());
+
+        /* Example JSON request:
+         * {
+         *  "format" : "xml",
+         *  "cursor" : "1234",
+         *  "node" : {
+         *   "kind" : "diskread",
+         *   "fileName": "examplefilename",
+         *   "keyfilter" : "f1='1    '",
+         *   "choosen" : "5",
+         *   "cursor" : "12345", // cursor handle
+         *   "input" : {
+         *    "f1" : "string5",
+         *    "f2" : "string5"
+         *   },
+         *   "output" : {
+         *    "f2" : "string",
+         *    "f1" : "real"
+         *   }
+         *  }
+         * }
+         *
+         */
+
+        const char *outputFmtStr = requestTree->queryProp("format");
+        int cursorHandle = requestTree->getPropInt("cursor");
+        Owned<IPropertyTree> responseTree; // Used if xml/json
+        OutputFormat outputFormat;
+        if (!outputFmtStr || strieq("xml", outputFmtStr))
+            outputFormat = outFmt_Xml;
+        else if (strieq("json", outputFmtStr))
+            outputFormat = outFmt_Json;
+        else
+            outputFormat = outFmt_Binary;
+        if (outFmt_Binary != outputFormat)
+            responseTree.setown(createPTree("Response"));
+
+        Owned<IRemoteActivity> outputActivity;
+        OpenFileInfo fileInfo;
+        if (!cursorHandle)
+        {
+            // In future this may be passed the request and build a chain of activities and return sink.
+            outputActivity.setown(createOutputActivity(*requestTree));
+
+            MemoryBuffer cursorMb;
+            if (requestTree->getPropBin("cursorBin", cursorMb))
+            {
+                cursorMb.setEndian(BIG_ENDIAN);
+                outputActivity->restoreCursor(cursorMb);
+            }
+
+            StringBuffer requestStr("jsonrequest:");
+            outputActivity->getInfoStr(requestStr);
+            Owned<StringAttrItem> name = new StringAttrItem(requestStr);
+
+            CriticalBlock block(sect);
+            cursorHandle = getNextHandle();
+            client.previdx = client.openFiles.ordinality();
+            client.openFiles.append(OpenFileInfo(cursorHandle, outputActivity, name));
+        }
+        else if (!lookupFileIOHandle(cursorHandle, fileInfo))
+            cursorHandle = NotFound; // challenge response ..
+        else // known handle, continuation
+            outputActivity.set(fileInfo.activity);
+
+        if (outFmt_Binary != outputFormat)
+            responseTree->setPropInt("cursor", cursorHandle);
+        else
+            reply.append(cursorHandle);
+        if (NotFound != cursorHandle)
```

Comment:
again, here cursorHandle is checked against NotFound instead of 0.

Reply:
will change use of NotFound to 0 for consistency.